### PR TITLE
Generate CRDs

### PR DESCRIPTION
**Pull request by arjunrn**
Generate CRDs using tools

Updated:
* Set `preserveUnknownFields: false` such that unknown fields gives a validation error.
* Remove `x-kubernetes-list-map-keys:` which doesn't work: https://github.com/kubernetes-sigs/controller-tools/issues/444

### CODE REVIEW

File: pkg/apis/zalando.org/v1/types.go [hunk 1]
```diff
@@ -179,19 +187,25 @@ type ActualTraffic struct {
...
 
 // DesiredTraffic is the desired traffic setting to direct traffic to
 // a stack. This is meant to use by clients to orchestrate traffic
 // switching.
 type DesiredTraffic struct {
-	StackName string  `json:"stackName"`
-	Weight    float64 `json:"weight"`
+	StackName string `json:"stackName"`
+	// +kubebuilder:validation:Type=number
+	// +kubebuilder:validation:Format=float
```
Comment (muaazsaleem) [inline pkg/apis/zalando.org/v1/types.go:202]: nit: Is this deliberately not `float64`? i.e // +kubebuilder:validation:Format=float64

Comment (muaazsaleem) [reply pkg/apis/zalando.org/v1/types.go:202]: Ah nvm, I see [here](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#schemaObject) that it's not a valid format.

**Comment by muaazsaleem** [comment]
👍

**Comment by mikkeloscar** [comment]
:+1:

**Pull request merged**